src/components/MoneyTemporaryForRefactorRequestConfirmationList.js

```diff
@@ -368,7 +368,7 @@ function MoneyTemporaryForRefactorRequestConfirmationList({
 
     const optionSelectorSections = useMemo(() => {
         const sections = [];
-        const unselectedParticipants = _.filter(selectedParticipantsFiltered, (participant) => !participant.selected);
+        const unselectedParticipants = _.filter(selectedParticipants, (participant) => !participant.selected);
```

Comment:
My gosh, these names _are_ confusing. So we have...

- `selectedParticipants`
- `selectedParticipantsFiltered`, `filter`ed on   `participant.selected`
- `unselectedParticipants`, **also** `filter`ed, but on the negation of the above, i.e. on `!participant.selected`

Also, we consider `selectedParticipantsFiltered` heavy enough so it needs to be memoized, but `unselectedParticipants` **not** heavy enough to be memoized.

It's great that we found the root cause and fixed it, but maybe let's minimize the risk of a similar problem in this very area.

Do you understand the difference between `selectedParticipants` and `selectedParticipantsFiltered`? Which one is named misleadingly?

We need either some renaming, or some comments, or both

Reply:
@cubuspl42 

`selectedParticipants` are the participants selected in participants page(group chat members in our case)
`selectedParticipantsFiltered` are participants with `selected = true`. User can toggle them in the confirmation page. Initially all values of selected are true
`unselectedParticipants` are participants with `selected = false`

Reply:
So maybe let's use another term for the "outer" selected participants, like `chosenParticipants` or `pickedParticipants`?

Then we could name the "inner" selected participants symmetrically, like...

- `selectedParticipants` (where `selected = true`)
- `unselectedParticipants` (where `selected = false`)

I think it would be clearer to declare them close to each other.

It's important to think about why it was easy to make a mistake and what we can do so it's, well, at least a bit harder to make a similar one in the future.

I searched the code briefly and it seems that there are not many usages to fix.

Reply:
@cubuspl42 

Please check again and let me know your thoughts